lib/ansible/modules/windows/win_firewall.ps1

```diff
@@ -19,50 +19,51 @@
 # WANT_JSON
 # POWERSHELL_COMMON
 
+$ErrorActionPreference = "Stop"
 
-# get params
-$params = Parse-Args $args -supports_check_mode $false
+$params = Parse-Args $args -supports_check_mode $true
+$check_mode = Get-AnsibleParam -obj $params -name "_ansible_check_mode" -type "bool" -default $false
 
-$profiles = Get-AnsibleParam -obj $params -name "profiles" -type "list" -default [ "Public", "Domain", "Private" ]
-$wantedstate = Get-AnsibleParam -obj $params -name "state" -type "str" -failifempty $true -validateset 'enabled', 'disabled'
+$state = Get-AnsibleParam -obj $params -name "state" -type "str" -failifempty $true -validateset 'disabled','enabled'
+$profiles = Get-AnsibleParam -obj $params -name "profiles" -type "list" -default ["Domain", "Private", "Public"]
 
 $result = @{
     changed = $false
-
 }
 
 Try {
 
-  ForEach($profile in $profiles)
+  ForEach ($profile in @('Domain', 'Private', 'Public')) {
+
+    $currentstate = (Get-NetFirewallProfile -Name $profile).Enabled
+    $result.$profile = @{
+        enabled = ($currentstate -eq $true)
```

Comment:
would 
enabled = $currentstate
work here?  it would be easier to read and understand if so.